crates/bevy_ecs/src/system/mod.rs

```diff
@@ -38,6 +38,8 @@ mod tests {
     struct B;
     struct C;
     struct D;
+    struct E;
```

Comment:
I'd be tempted to use some macros and make structs `T0,T1,...,T16`. However, this approach is also reasonable.